coalib/results/Diff.py

```diff
@@ -12,17 +12,20 @@ class Diff:
     A Diff result represents a difference for one file.
     """
 
-    def __init__(self, file_list, rename=False, delete=False):
+    def __init__(self, file_list, filename=None, rename=False, delete=False):
         """
         Creates an empty diff for the given file.
 
         :param file_list: The original (unmodified) file as a list of its
                           lines.
+
+        :param filename:  None or a string containing the name of the file
         :param rename:    False or str containing new name of file.
         :param delete:    True if file is set to be deleted.
         """
         self._changes = {}
         self._file = file_list
+        self.filename = filename
```

Comment:
if filename isn't given, can we print a warning? (Note that before @Uran198 s PR about logging isn't merged we don't use builtin logging yet, add a todo to transform it to logging later?)